### PR TITLE
new package: openjdk-8 iced tea distribution

### PR DESCRIPTION
**Pull request by joshrwolf**
new package: openjdk-8

builds off the work done in #1681 to bootstrap openjdk-8 using the icedtea distribution

related: https://github.com/chainguard-dev/internal/issues/1575

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [x] REQUIRED - The package is added to `packages.txt`

### CODE REVIEW

File: openjdk-8.yaml [hunk 1]
```diff
@@ -0,0 +1,222 @@
+package:
+  name: openjdk-8
+  version: 8u372-b07
```
Comment (joshrwolf) [inline openjdk-8.yaml:3]: seems like this isn't parseable by `wolfictl`, but I'm not sure this is the version naming we want to use given this is based on icedtea? follow up to that is how the icedtea versions translate to the openjdk versions...

Comment (patflynn) [reply openjdk-8.yaml:3]: @rawlingsj may have advice.

File: openjdk-8.yaml [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  epoch: 0
+  description: "IcedTea distribution of OpenJDK 8 (UNSUPPORTED)"
```
Comment (patflynn) [inline openjdk-8.yaml:5]: We do want to support JDK 8 since it's still supported upstream

File: openjdk-8.yaml [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+      rm -f "${{targets.destdir}}"/usr/lib/jvm/java-1.8-openjdk/server/classes.jsa
+
+      # symlink to shared java cacerts store
+      rm -f "${{targets.destdir}}"/usr/lib/jvm/jvm-1.8-openjdk/jre/lib/security/cacerts
+      ln -sf /etc/ssl/certs/java/cacerts \
+        "${{targets.destdir}}"/usr/lib/jvm/java-1.8-openjdk/jre/lib/security/cacerts
+
+  - uses: strip
+
+subpackages:
```
Comment (patflynn) [inline openjdk-8.yaml:137]: @kaniini I think will have suggestions related to -default-jvm

File: openjdk-8.yaml [hunk 4]
```diff
@@ -0,0 +1,222 @@
+package:
+  name: openjdk-8
```
Comment (patflynn) [inline openjdk-8.yaml:2]: should we call out the icedtea provenance? @kaniini

File: openjdk-8.yaml (hunk 2 above)
Comment (joshrwolf) [inline openjdk-8.yaml:5]:
```suggestion
  description: "IcedTea distribution of OpenJDK 8"
```

File: openjdk-8.yaml (hunk 3 above)
Comment (joshrwolf) [reply openjdk-8.yaml:137]: the default-jvm for 7 was done as a subpackage, I can do the equivalent here if that's how we want to keep things!

https://github.com/wolfi-dev/os/blob/main/openjdk-7.yaml#L267-L277

File: openjdk-8.yaml (hunk 4 above)
Comment (kaniini) [reply openjdk-8.yaml:2]: We do in the description.

File: openjdk-8.yaml (hunk 3 above)
Comment (kaniini) [reply openjdk-8.yaml:137]: Yes, that's how all JVMs should be packaged in Wolfi moving forward.  We will do similar for dotnet when 6 lands.

File: openjdk-8.yaml (hunk 1 above)
Comment (kaniini) [reply openjdk-8.yaml:3]: This should probably be 8.372.07.

File: openjdk-8.yaml [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+      - krb5-dev
+      - pkgconf-dev
+      - attr-dev
+      - expat-dev
+      - fontconfig-dev
+      - libxi-dev
+      - libjpeg-dev
+      - giflib-dev
+      - lcms2-dev
+      - java-gcj-compat-default-jvm
```
Comment (kaniini) [inline openjdk-8.yaml:51]: This should probably be openjdk-7-default-jvm.

```suggestion
      - openjdk-7-default-jvm
```

**Review by kaniini — APPROVED** [review]

**Pull request merged**